stellargraph/core/graph.py

```diff
@@ -525,25 +535,29 @@ def edges(
                 separate list.
 
         Returns:
-            The graph edges. If edge weights are included then a tuple of (edges, weights)
+            The graph edges. If edge weights are included then a tuple of (edges, weights).
         """
-        # FIXME: these would be better returned as the 2 or 3 arrays directly, rather than tuple-ing
-        # (the same applies to all other instances of zip in this file)
-        if include_edge_type:
-            edges = list(
-                zip(
-                    self._edges.sources,
-                    self._edges.targets,
-                    self._edges.type_of_iloc(slice(None)),
-                )
-            )
-        else:
-            edges = list(zip(self._edges.sources, self._edges.targets))
-
+        edge_arrs = self.edge_arrays(include_edge_type, include_edge_weight)
+        edges = list(zip(*(arr for arr in edge_arrs[:3] if arr is not None)))
         if include_edge_weight:
-            return edges, self._edges.weights
+            return edges, edge_arrs[3]
+        return self._to_edges(edge_arrs)
 
-        return edges
+    def edge_arrays(self, include_edge_type=False, include_edge_weight=False) -> tuple:
+        """
+        Obtains the collection of edges in the graph.
+
+        Args:
+            include_edge_type (bool): A flag that indicates whether to return edge types.
+            include_edge_weight (bool): A flag that indicates whether to return edge weights.
+
+        Returns:
+            A tuple containing 1D arrays of the source and target nodes (sources, targets). Setting include_edge_type
+                and/or include_edge_weight to True will include arrays of edge types and/or edge weights in this tuple.
```

Comment:
I think this would be a bit more accurate, since we're always returning a 4-tuple in a particular order, so it'd be nice to spell out the order things are in:
```suggestion
        Obtains the collection of edges in the graph as a tuple of arrays (sources, targets, types, weights). ``types`` and ``weights`` will be `None` if the optional parameters are not specified.

        Args:
            include_edge_type (bool): A flag that indicates whether to return edge types.
            include_edge_weight (bool): A flag that indicates whether to return edge weights.

        Returns:
            A tuple containing 1D arrays of the source and target nodes (sources, targets, types, weights). Setting include_edge_type
                and/or include_edge_weight to True will include arrays of edge types and/or edge weights in this tuple, otherwise they will be set to ``None``.
```